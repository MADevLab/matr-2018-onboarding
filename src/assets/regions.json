[
  {
    "id": "NE",
    "title": "Northeast",
    "regions": [
      "CT",
      "DC",
      "DE",
      "MA",
      "MD",
      "ME",
      "NH",
      "NJ",
      "NY",
      "PA",
      "PR",
      "RI",
      "VT"
    ]
  },
  {
    "id": "MW",
    "title": "Midwest",
    "regions": [
      "IA",
      "IL",
      "IN",
      "KS",
      "MI",
      "MN",
      "MO",
      "ND",
      "NE",
      "OH",
      "SD",
      "WI"
    ]
  },
  {
    "id": "SO",
    "title": "South",
    "regions": [
      "AL",
      "AR",
      "FL",
      "GA",
      "KY",
      "LA",
      "MS",
      "NC",
      "OK",
      "SC",
      "TN",
      "TX",
      "VA",
      "WV"
    ]
  },
  {
    "id": "WE",
    "title": "West",
    "regions": [
      "AK",
      "AZ",
      "CA",
      "CO",
      "HI",
      "ID",
      "MT",
      "NM",
      "NV",
      "OR",
      "UT",
      "WA",
      "WY"
    ]
  }
]
