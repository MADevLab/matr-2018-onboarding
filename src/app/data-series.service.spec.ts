import { HttpClientModule } from '@angular/common/http';
import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
import { TestBed, inject } from '@angular/core/testing';

import { DataSeriesService } from './data-series.service';
import { DataSet } from './data-set.model';
import { DataSetOption } from './data-set-option.model';
import { Regions } from './regions.model';

describe('DataSeriesService', () => {
  let service: DataSeriesService;
  let httpMock: HttpTestingController;

  beforeEach(() => {
    TestBed.configureTestingModule({
      imports: [HttpClientTestingModule],
      providers: [DataSeriesService]
    });

    service = TestBed.get(DataSeriesService);
    httpMock = TestBed.get(HttpTestingController);
  });

  afterEach(() => {
    httpMock.verify();
  });

  it('is defined', () => {
    expect(service).toBeDefined();
  });

  it('gets the list of data sets', () => {
    let dataSets: Map<number, DataSetOption>;
    const sub = service.getDataSets().subscribe(ds => dataSets = ds);

    sub.unsubscribe();

    expect(dataSets).toMatchSnapshot();
  });

  it('gets the regions', () => {
    let regions: Regions[];
    const sub = service.getRegions().subscribe(r => regions = r);
    const req = httpMock.expectOne('assets/regions.json');

    expect(req.request.method).toBe('GET');
    req.flush(_regions);

    expect(regions).toMatchSnapshot();
  });

  it('gets a data set', () => {
    let dataSet: DataSet;
    const sub = service.getDataSet(1).subscribe(d => dataSet = d);
    const req = httpMock.expectOne(`${service.s3Url}/gross-domestic-product.json`);

    expect(req.request.method).toBe('GET');
    req.flush(_dataSet);

    expect(dataSet).toMatchSnapshot();
  });
});

const _regions = [
  {
    id: 'NE',
    title: 'Northeast',
    regions: ['CT', 'DC', 'DE']
  },
  {
    id: 'MW',
    title: 'Midwest',
    regions: ['WI']
  },
  {
    id: 'SO',
    title: 'South',
    regions: ['AL', 'AR']
  },
  {
    id: 'WE',
    title: 'West',
    regions: ['WA', 'WY']
  }
];

const _dataSet = {
  concept: 'FGDP$',
  description: 'Gross State Product: Total, (Bil. Chained 2009 $, SAAR)',
  source: 'U.S. Bureau of Economic Analysis (BEA); Moody\'s Analytics Estimated and Forecasted',
  freq: 162,
  transformation: 'None',
  data: [
    {
      period: 661,
      date: '2015Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 662,
      date: '2015Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 664,
      date: '2015Q4',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 663,
      date: '2015Q3',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 665,
      date: '2016Q1',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    },
    {
      period: 666,
      date: '2016Q2',
      mapData: [
        {
          value: 49.29,
          code: 'ak'
        },
        {
          value: 177.56,
          code: 'al'
        },
        {
          value: 107.06,
          code: 'ar'
        },
        {
          value: 263.13,
          code: 'az'
        },
        {
          value: 2226.61,
          code: 'ca'
        },
        {
          value: 286.63,
          code: 'co'
        },
        {
          value: 225.09,
          code: 'ct'
        }
      ]
    }
  ]
};
