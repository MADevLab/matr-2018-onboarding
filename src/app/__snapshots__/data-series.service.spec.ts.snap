// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataSeriesService gets a data set 1`] = `
Object {
  "concept": "FGDP$",
  "data": Array [
    Object {
      "date": "2015Q1",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 661,
    },
    Object {
      "date": "2015Q2",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 662,
    },
    Object {
      "date": "2015Q4",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 664,
    },
    Object {
      "date": "2015Q3",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 663,
    },
    Object {
      "date": "2016Q1",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 665,
    },
    Object {
      "date": "2016Q2",
      "mapData": Array [
        Object {
          "code": "ak",
          "value": 49.29,
        },
        Object {
          "code": "al",
          "value": 177.56,
        },
        Object {
          "code": "ar",
          "value": 107.06,
        },
        Object {
          "code": "az",
          "value": 263.13,
        },
        Object {
          "code": "ca",
          "value": 2226.61,
        },
        Object {
          "code": "co",
          "value": 286.63,
        },
        Object {
          "code": "ct",
          "value": 225.09,
        },
      ],
      "period": 666,
    },
  ],
  "description": "Gross State Product: Total, (Bil. Chained 2009 $, SAAR)",
  "freq": 162,
  "source": "U.S. Bureau of Economic Analysis (BEA); Moody's Analytics Estimated and Forecasted",
  "transformation": "None",
}
`;

exports[`DataSeriesService gets the list of data sets 1`] = `
Map {
  0 => Object {
    "dataUrl": "home-sales.json",
    "label": "home sales",
  },
  1 => Object {
    "dataUrl": "gross-domestic-product.json",
    "label": "gross domestic product",
  },
  2 => Object {
    "dataUrl": "avg-household-income.json",
    "label": "avg. household income",
  },
}
`;

exports[`DataSeriesService gets the regions 1`] = `
Array [
  Object {
    "id": "NE",
    "regions": Array [
      "CT",
      "DC",
      "DE",
    ],
    "title": "Northeast",
  },
  Object {
    "id": "MW",
    "regions": Array [
      "WI",
    ],
    "title": "Midwest",
  },
  Object {
    "id": "SO",
    "regions": Array [
      "AL",
      "AR",
    ],
    "title": "South",
  },
  Object {
    "id": "WE",
    "regions": Array [
      "WA",
      "WY",
    ],
    "title": "West",
  },
]
`;
